draft-ietf-rats-corim.md

```diff
@@ -1207,6 +1207,242 @@ The following describes each member of the `concise-bom-tag` map.
   The `$$concise-bom-tag-extension` extension socket is empty in this
   specification.
 
+# CoRIM-based Evidence Verification
+
+The verification procedure is divided into three separate phases:
+
+* Appraisal Context initialisation
+* Evidence collection
+* Evidence appraisal
+
+At a few well-defined points in the procedure, the Verifier behaviour will
+depend on the specific CoRIM profile.
+Each CoRIM profile MUST provide a description of the expected Verifier behavior
+for each of those well-defined points.
+
+Note that what follows describes a simplified implementation.
+Verifiers claiming compliance with this specification MUST exhibit the same
+externally visible behavior as described here,
+they are not required to use the same internal data structures.
+For example, it is expected that the resources used during the initialisation
+phase can be amortised across multiple appraisals.
+
+## Appraisal Context initialisation
+
+The goal of the initialisation phase is to load the CoRIM Appraisal Context
+with objects such as tags (CoMID, CoSWID etc.) from CoRIM files,
+cryptographic validation key material (e.g., raw public keys, root certificates,
+intermediate CA certificate chains), etc. that will be used in the subsequent
+Evidence Appraisal phase.
+
+### CoRIM Selection
+
+All available CoRIMs are collected.
+A Verifier may be pre-configured with a large number of tags describing many
+types of device.
+All CoRIMs are loaded at this stage, later stages will select the CoRIMs
+appropriate to the Evidence Appraisal step.
+
+CoRIMs that are not within their validity period, or that cannot be associated
+with an authenticated and authorised source MUST be discarded.
+
+CoRIM that are secured by a cryptographic mechanism such as a signature which
+does not pass validation MUST be discarded.
+
+Other selection criteria MAY be applied.
+
+For example, if the Evidence format is known in advance, tags that do not match
+the expected profile can be readily discarded.
+
+The selection process MUST yield at least one usable tag.
+
+### CoBOM Extraction
+
+All the available Concise Bill Of Material (CoBOMs) tags are then collected
```

Comment:
Can you please help me clarify:

For a given profile, only one active CoBOM is expected at the point of verification.

So does this section assumes that the set of CoBOMs pertain to all the CoBOMs (for a variety of attester types) and thus all profiles which Verifier handles?
If such is the assumption, then one must explain how to filter the `active` CoBOM relevant to a specific profile.

Reply:
CoBOMs can contain tags that are not in the RIM that immediately contains the CoBOM. Dependent RIMS (containing these tags) may, themselves, have CoBOMs that reference tags in the immediate RIM (or yet other RIMS), and so forth. 
Presumably, there is a "top level" CoBOM that correctly describes the "device" that is being attested. The question for Verifiers is how does it know it has the complete description (via tags) of the device? Which CoBOM is the 'root' CoBOM?